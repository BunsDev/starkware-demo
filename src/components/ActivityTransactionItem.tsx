import React from 'react';
import { Grid, styled, Typography } from '@material-ui/core';
import { Transaction, TransactionType } from 'context/user';
import mintIcon from 'assets/icons/mint-icon.svg';
import swapIcon from 'assets/icons/swap-icon.svg';

const ActivityListItem = styled('div')(({ theme }) => ({
	height: '81px',
	width: '100%',
	display: 'flex',
	justifyContent: 'space-between',
	boxSizing: 'border-box',
	borderTop: '1px solid rgb(193,193,255,0.2)!important',
	'& div': {
		fontFamily: 'IBM Plex Sans',
		fontStyle: 'normal',
	},
}));
const ActivityDescription = styled('div')(({ theme }) => ({
	height: '80px',
	width: 'fit-content',
	justifyContent: 'flex-start',
	display: 'flex',
	flexDirection: 'row',
}));
const ActivityValue = styled('div')(({ theme }) => ({
	width: 'fit-content',
}));
const ActivityIcon = styled('img')({
	height: '32px',
	width: '32px',
});
const ActivityType = styled('div')({
	height: 'fit-content',
	marginBottom: '13px',

	fontWeight: 'normal',
	fontSize: '18px',
	color: '#FAFAF5',
});
const ActivityDate = styled('div')({
	fontSize: '14px',
	color: '#FAFAF5',
	opacity: 0.6,
});
const Flex = styled('div')({
	display: 'flex',
	justifyContent: 'start',
	alignItems: 'center',
	marginRight: '16px',
});
const ColumnFlex = styled('div')({
	display: 'flex',
	justifyContent: 'center',
	flexDirection: 'column',
	height: '80px',
});

interface ActivityTransactionProps {
	transaction: Transaction;
}
export const ActivityTransactionItem: React.FC<ActivityTransactionProps> = ({
	transaction,
	...props
}) => {
	return (
		<ActivityListItem>
			<ActivityDescription>
				<Flex>
					<ActivityIcon src={transaction.type === TransactionType.MINT ? mintIcon : swapIcon} />
				</Flex>
				<ColumnFlex>
					<ActivityType>{transaction.displayName}</ActivityType>
					<ActivityDate>{transaction.executedOn}</ActivityDate>
				</ColumnFlex>
			</ActivityDescription>
			<ColumnFlex>
				<ActivityValue>{`+${transaction.value} ${transaction.incomingToken.name}`}</ActivityValue>
			</ColumnFlex>
		</ActivityListItem>
	);
};
