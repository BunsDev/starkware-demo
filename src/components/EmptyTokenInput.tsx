import { Button, createStyles, Grid, makeStyles, styled } from '@material-ui/core';
import React from 'react';
import { DarkBox } from './common/DarkBox';
import { TokenIcon } from './common/TokenIcon';
import { NumericInput } from './NumericInput';
import { ReactComponent as TokenSVG } from 'assets/tokens/token.svg';

const InputsContainer = styled(Grid)({
	padding: '0 0 0 12px',
});

const useButtonStyles = makeStyles(() =>
	createStyles({
		root: {
			borderRadius: 50,
		},
		label: {
			fontSize: 12,
      maxWidth: 95,
      height: 16,
		},
	}),
);

export const EmptyTokenInput: React.FC = () => {
  const buttonStyles = useButtonStyles()

	return (
		<DarkBox>
			<Grid container alignItems="center">
				<Grid item>
					<TokenIcon Icon={TokenSVG} size="large" />
				</Grid>
				<InputsContainer item xs>
					<Grid container justify="space-between" alignItems="center">
						<Grid item xs>
							<Button color="secondary" variant="outlined" classes={buttonStyles}>
								Select a Token
							</Button>
						</Grid>
						<Grid item xs>
							<NumericInput onChange={() => false} />
						</Grid>
					</Grid>
				</InputsContainer>
			</Grid>
		</DarkBox>
	);
};
