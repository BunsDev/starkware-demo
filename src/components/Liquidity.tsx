import React from 'react';

export const Liquidity = (): JSX.Element => {
	return <h1>Liquidity</h1>;
};
