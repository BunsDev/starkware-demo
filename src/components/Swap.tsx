import React, { useState, useCallback, useMemo } from 'react';
import { Button, Grid, IconButton, styled, Typography } from '@material-ui/core';
import { BigNumber } from '@ethersproject/bignumber';
import { ReactComponent as SwapDirection } from 'assets/icons/swap-direction.svg';
import { TokenSelector } from './TokenSelector';
import { DarkBox } from './common/DarkBox';
import { NumericInput } from './NumericInput';
import { Token, TokenBalance } from 'services/API/token/types';
import { useTokens } from 'services/API/token/hooks/useTokens';
import { useTokenOptions } from 'services/API/token/hooks/useTokenOptions';
import { useTokenBalances } from 'services/API/token/hooks/useTokenBalances';
import { RoundedButton } from './common/RoundedButton';

const StyledInputContainer = styled(Grid)({
	padding: '0 0 0 12px',
});

const StyledArrowsContainer = styled(Grid)({
	width: '100%',
	margin: '0 auto -23px auto',
	height: 65,
});

const StyledSwapButton = styled(Button)(({ theme }) => ({
	marginTop: 32,
	height: 66,
	background: 'rgba(145, 145, 183, 0.15);',
	color: theme.palette.text.primary,
	borderColor: 'rgba(145, 145, 183, 0.15);',
	borderRadius: 6,
	boxShadow: 'unset',
	'&:hover': {
		backgroundColor: 'rgba(145, 145, 183, 0.1);',
		boxShadow: 'unset',
	},
}));

const StyledLeftLabelText = styled(Typography)({
	fontWeight: 600,
});

const StyledRightLabelText = styled(Typography)({
	fontWeight: 400,
});

const LabelsContainer = styled(Grid)({
	paddingBottom: '5px',
});

const Labels = ({ leftText, rightText }: { leftText: string; rightText: string }): JSX.Element => (
	<LabelsContainer container justify="space-between">
		<Grid item>
			<StyledLeftLabelText variant="subtitle1" color="textPrimary">
				{leftText}
			</StyledLeftLabelText>
		</Grid>
		<Grid item>
			<StyledRightLabelText variant="subtitle1" color="textSecondary">
				{rightText}
			</StyledRightLabelText>
		</Grid>
	</LabelsContainer>
);

export const Swap = (): JSX.Element => {
	const [fromValue, setFromValue] = useState('');
	const [toValue, setToValue] = useState('');
	const { data: tokensData } = useTokens();
	const [fromToken, setFromToken] = useState<Token | undefined>();
	const [fromAmount, setFromAmount] = useState<string>();
	const [toToken, setToToken] = useState<Token>();
	const [toAmount, setToAmount] = useState<string>();
	const { data: tokenBalances } = useTokenBalances();

	const conversionRates = useMemo(() => {
		if (fromToken && toToken) {
			return {
				from: BigNumber.from(toToken.price).div(BigNumber.from(fromToken.price)),
				to: BigNumber.from(fromToken.price).div(BigNumber.from(toToken.price)),
			};
		}

		return {
			from: 1,
			to: 1,
		};
	}, [fromToken, toToken]);

	let fromBalance: TokenBalance | undefined;

	if (fromToken && tokenBalances) {
		fromBalance = tokenBalances.find(
			(tokenBalance) => tokenBalance.token.symbol === fromToken.symbol,
		);
	}

	let toBalance: TokenBalance | undefined;

	if (toToken && tokenBalances) {
		toBalance = tokenBalances.find((tokenBalance) => tokenBalance.token.symbol === toToken.symbol);
	}

	const handleSwitch = useCallback(() => {
		setFromValue(toValue);
		setToValue(fromValue);

		setFromToken(toToken);
		setToToken(fromToken);

		setFromAmount(toAmount);
		setToAmount(fromAmount);
	}, [fromAmount, fromToken, fromValue, toAmount, toToken, toValue]);

	const handleFromTokenSelected = (token: Token) => {
		if (toToken && token.symbol === toToken.symbol) {
			setToToken(undefined);
		}

		setFromAmount('0');
		setFromToken(token);
	};

	const handleFromAmountChange = (amount: string) => {
		setFromAmount(amount);

		if (conversionRates) {
			setToAmount(BigNumber.from(amount).mul(BigNumber.from(conversionRates.to)).toString());
		}
	};

	const handleToAmountChange = (amount: string) => {
		setToAmount(amount);

		if (conversionRates) {
			setFromAmount(BigNumber.from(amount).mul(BigNumber.from(conversionRates.from)).toString());
		}
	};

	const options = useTokenOptions(fromToken, tokensData);

	return (
		<Grid container>
			<Grid item xs={12}>
				<Labels leftText="From" rightText={fromBalance ? `Balance: ${fromBalance.amount}` : ''} />
				<DarkBox>
					<Grid container alignItems="center">
						<Grid item xs aria-label="token to swap">
							<TokenSelector
								value={fromToken}
								options={options}
								onChange={handleFromTokenSelected}
							/>
						</Grid>
						{fromToken && (
							<StyledInputContainer item xs>
								<Grid container justify="flex-end" alignItems="center">
									<Grid item xs={6}>
										<NumericInput
											inputProps={{
												'aria-label': 'amount of token to swap',
											}}
											value={fromAmount}
											handleChange={(change) => handleFromAmountChange(change)}
										/>
									</Grid>
									{fromBalance && (
										<Grid item xs={6}>
											<RoundedButton
												onClick={() => handleFromAmountChange((fromBalance as TokenBalance).amount)}
											>
												Max
											</RoundedButton>
										</Grid>
									)}
								</Grid>
							</StyledInputContainer>
						)}
					</Grid>
				</DarkBox>
			</Grid>
			<Grid item xs={12}>
				<StyledArrowsContainer container justify="center" alignItems="center">
					<Grid item>
						<IconButton aria-label="invert tokens swap direction" onClick={handleSwitch}>
							<SwapDirection />
						</IconButton>
					</Grid>
				</StyledArrowsContainer>
			</Grid>
			<Grid item xs={12}>
				<Labels leftText="To" rightText={toBalance ? `Balance: ${toBalance.amount}` : ''} />
				<DarkBox>
					<Grid container alignItems="center">
						<Grid item xs aria-label="token to be swapped">
							<TokenSelector
								value={toToken}
								options={options}
								onChange={(token) => setToToken(token)}
							/>
						</Grid>
						{toToken && (
							<StyledInputContainer item xs>
								<Grid container justify="flex-end" alignItems="center">
									<NumericInput
										inputProps={{
											'aria-label': 'amount of token to be swapped',
										}}
										value={toAmount}
										handleChange={(change) => handleToAmountChange(change)}
									/>
								</Grid>
							</StyledInputContainer>
						)}
					</Grid>
				</DarkBox>
			</Grid>
			<Grid item xs={12}>
				<StyledSwapButton fullWidth variant="contained">
					Swap
				</StyledSwapButton>
			</Grid>
		</Grid>
	);
};
