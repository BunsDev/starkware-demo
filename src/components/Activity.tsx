import React, { useContext, useMemo } from 'react';
import { Grid, styled } from '@material-ui/core';
import { UserContext, TransactionType } from 'context/user';
import { ActivityTransactionItem } from 'components/ActivityTransactionItem';

const ActivityListWrapper = styled('div')({
	maxHeight: '431px',
	height: 'fit-content',
	width: '100%',
	overflowY: 'auto',
});

export const Activity = (): JSX.Element => {
	const { state: userState } = useContext(UserContext);
	const activity = userState.activity;

	const formattedActivity = useMemo(() => {
		return activity
			.map((transaction) => {
				if (transaction.type === TransactionType.MINT) {
					if (transaction.args.mint2) {
						return [
							{
								type: transaction.type,
								timestamp: transaction.timestamp,
								amount: transaction.args.mint1.amount,
								symbol: transaction.args.mint1.token.symbol,
							},
							{
								type: transaction.type,
								timestamp: transaction.timestamp,
								amount: transaction.args.mint2.amount,
								symbol: transaction.args.mint2.token.symbol,
							},
						];
					}

					return {
						type: transaction.type,
						timestamp: transaction.timestamp,
						amount: transaction.args.mint1.amount,
						symbol: transaction.args.mint1.token.symbol,
					};
				}

				return {
					type: transaction.type,
					timestamp: transaction.timestamp,
					amount: transaction.args.to.amount,
					symbol: transaction.args.to.token.symbol,
				};
			})
			.flat();
	}, [activity]);

	return (
		<Grid container>
			<Grid item xs={12}>
				<ActivityListWrapper>
					{formattedActivity.map((transaction, i) => (
						<ActivityTransactionItem key={`transaction-${i}`} {...transaction} />
					))}
				</ActivityListWrapper>
			</Grid>
		</Grid>
	);
};
