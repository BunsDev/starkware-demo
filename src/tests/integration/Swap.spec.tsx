import React from 'react';
import '@testing-library/jest-dom';

import { customRender, screen, fireEvent, within, waitForElementToBeRemoved } from '../utils';
import { Swap } from '../../components/Swap';

describe('Swap', () => {
	// Select both "From" and "To" tokens
	const setupSwapTokens = async () => {
		const { container } = customRender(<Swap />);
		fireEvent.click(screen.getByRole('button', { name: /Select a Token/i }));
		// TODO: replace name with actual tokens from useQuery
		const dialog = screen.getByRole('presentation');
		fireEvent.click(within(dialog).getByText(/TK2/i));
		await waitForElementToBeRemoved(dialog); // close dialog
		return container;
	};

	it('displays "From Token" with a default token', () => {
		const { container } = customRender(<Swap />);
		expect(container).toMatchSnapshot();
	});

	it('can select a "To Token"', async () => {
		const container = await setupSwapTokens();
		expect(container).toMatchSnapshot();
	});

	it('does not display the already selected "From Token" as option if it is tried to be changed', () => {
		customRender(<Swap />);
		fireEvent.click(screen.getByRole('button', { name: /change token/i }));
		expect(screen.getByRole('presentation')).toMatchSnapshot();
	});

	it('can tokens swap direction', async () => {
		const container = await setupSwapTokens();
		fireEvent.click(screen.getByRole('button', { name: /invert tokens swap direction/i }));
		expect(container).toMatchSnapshot();
	});

	it('can change swap tokens amounts', async () => {
		const container = await setupSwapTokens();
		fireEvent.change(screen.getByLabelText(/amount of token to swap/i), {
			target: { value: '128' },
		});
		fireEvent.change(screen.getByLabelText(/amount of token to be swapped/i), {
			target: { value: '256' },
		});
		expect(container).toMatchSnapshot();
	});
});
