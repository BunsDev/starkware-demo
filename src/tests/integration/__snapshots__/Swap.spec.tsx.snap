// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swap can change tokens amounts 1`] = `
<div>
  <div
    class="MuiGrid-root-1 MuiGrid-container-2"
  >
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-104 MuiGrid-container-2 MuiGrid-justify-xs-space-between-21"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <h6
            class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-105 MuiTypography-subtitle1-117 MuiTypography-colorTextPrimary-131"
          >
            From
          </h6>
        </div>
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <h6
            class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-136 MuiTypography-subtitle1-117 MuiTypography-colorTextSecondary-132"
          />
        </div>
      </div>
      <div
        class="MuiBox-root-138 MuiBox-root-139 Styled(MuiBox)-root-137"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-align-items-xs-center-10"
        >
          <div
            aria-label="token to swap"
            class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-item-3"
            >
              <div
                class="MuiBox-root-138 MuiBox-root-141 Styled(MuiBox)-root-140"
              >
                <div
                  class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-spacing-xs-1-24 MuiGrid-align-items-xs-center-10"
                >
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3"
                  >
                    <div
                      class="MuiAvatar-root-144 MuiAvatar-circle-146 WithStyles(ForwardRef(Avatar))-root-142 WithStyles(ForwardRef(Avatar))-root-143 MuiAvatar-colorDefault-145"
                      dimension="48"
                    >
                      <img
                        alt="token icon"
                        src="token.svg"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
                  >
                    <div
                      class="MuiBox-root-138 MuiBox-root-154 Styled(MuiBox)-root-152 Styled(MuiBox)-root-153"
                    >
                      <p
                        class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-155 MuiTypography-body1-108 MuiTypography-colorTextPrimary-131"
                      >
                        TK1
                      </p>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-spacing-xs-1-24 MuiGrid-item-3 MuiGrid-align-items-xs-center-10 MuiGrid-justify-xs-flex-end-20 MuiGrid-grid-xs-4-39 MuiGrid-grid-sm-6-55"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47 MuiGrid-grid-sm-6-55"
            >
              <div
                autocorrect="off"
                class="MuiInputBase-root-157  makeStyles-input-156 MuiInputBase-fullWidth-167"
                inputmode="decimal"
                spellcheck="false"
              >
                <input
                  aria-label="amount of token to swap"
                  autocomplete="off"
                  class="MuiInputBase-input-168"
                  pattern="^[0-9]*[.,]?[0-9]*$"
                  placeholder="0"
                  type="text"
                  value="100"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-175 MuiGrid-container-2 MuiGrid-align-items-xs-center-10 MuiGrid-justify-xs-center-19"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <button
            aria-label="invert tokens swap direction"
            class="MuiButtonBase-root-185 MuiIconButton-root-176"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label-184"
            >
              <svg>
                swap-direction.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root-259"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-104 MuiGrid-container-2 MuiGrid-justify-xs-space-between-21"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <h6
            class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-105 MuiTypography-subtitle1-117 MuiTypography-colorTextPrimary-131"
          >
            To
          </h6>
        </div>
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <h6
            class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-136 MuiTypography-subtitle1-117 MuiTypography-colorTextSecondary-132"
          />
        </div>
      </div>
      <div
        class="MuiBox-root-138 MuiBox-root-188 Styled(MuiBox)-root-137"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-align-items-xs-center-10"
        >
          <div
            aria-label="token to be swapped"
            class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-item-3"
            >
              <div
                class="MuiBox-root-138 MuiBox-root-189 Styled(MuiBox)-root-140"
              >
                <div
                  class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-spacing-xs-1-24 MuiGrid-align-items-xs-center-10"
                >
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3"
                  >
                    <div
                      class="MuiAvatar-root-144 MuiAvatar-circle-146 WithStyles(ForwardRef(Avatar))-root-142 WithStyles(ForwardRef(Avatar))-root-190 MuiAvatar-colorDefault-145"
                      dimension="48"
                    >
                      <img
                        alt="token icon"
                        src="token2.svg"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
                  >
                    <div
                      class="MuiBox-root-138 MuiBox-root-192 Styled(MuiBox)-root-152 Styled(MuiBox)-root-191"
                    >
                      <p
                        class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-155 MuiTypography-body1-108 MuiTypography-colorTextPrimary-131"
                      >
                        TK2
                      </p>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-align-items-xs-center-10 MuiGrid-grid-xs-4-39 MuiGrid-grid-sm-6-55"
          >
            <div
              autocorrect="off"
              class="MuiInputBase-root-157  makeStyles-input-156 MuiInputBase-fullWidth-167"
              inputmode="decimal"
              spellcheck="false"
            >
              <input
                aria-label="amount of token to be swapped"
                autocomplete="off"
                class="MuiInputBase-input-168"
                pattern="^[0-9]*[.,]?[0-9]*$"
                placeholder="0"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-193 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <h6
        class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-136 Styled(WithStyles(ForwardRef(Typography)))-root-194 MuiTypography-subtitle1-117 MuiTypography-colorTextSecondary-132"
      >
        <span>
          1 TK1 = 
        </span>
        <span>
          <span
            class="MuiSkeleton-root-196 MuiSkeleton-text-197 WithStyles(ForwardRef(Skeleton))-root-195 MuiSkeleton-pulse-200"
            style="width: 20px;"
          />
        </span>
        <span>
           TK2
        </span>
      </h6>
    </div>
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <button
        class="MuiButtonBase-root-185 MuiButton-root-208 MuiButton-contained-216 WithStyles(ForwardRef(Button))-root-207 MuiButton-containedSecondary-218 MuiButton-disableElevation-219 MuiButton-disabled-221 MuiButton-fullWidth-231 MuiButtonBase-disabled-186"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label-209"
        >
          Enter an amount
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Swap can swap tokens direction 1`] = `
<div>
  <div
    class="MuiGrid-root-1 MuiGrid-container-2"
  >
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-104 MuiGrid-container-2 MuiGrid-justify-xs-space-between-21"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <h6
            class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-105 MuiTypography-subtitle1-117 MuiTypography-colorTextPrimary-131"
          >
            From
          </h6>
        </div>
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <h6
            class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-136 MuiTypography-subtitle1-117 MuiTypography-colorTextSecondary-132"
          />
        </div>
      </div>
      <div
        class="MuiBox-root-138 MuiBox-root-139 Styled(MuiBox)-root-137"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-align-items-xs-center-10"
        >
          <div
            aria-label="token to swap"
            class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-item-3"
            >
              <div
                class="MuiBox-root-138 MuiBox-root-141 Styled(MuiBox)-root-140"
              >
                <div
                  class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-spacing-xs-1-24 MuiGrid-align-items-xs-center-10"
                >
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3"
                  >
                    <div
                      class="MuiAvatar-root-144 MuiAvatar-circle-146 WithStyles(ForwardRef(Avatar))-root-142 WithStyles(ForwardRef(Avatar))-root-143 MuiAvatar-colorDefault-145"
                      dimension="48"
                    >
                      <img
                        alt="token icon"
                        src="token2.svg"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
                  >
                    <div
                      class="MuiBox-root-138 MuiBox-root-154 Styled(MuiBox)-root-152 Styled(MuiBox)-root-153"
                    >
                      <p
                        class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-155 MuiTypography-body1-108 MuiTypography-colorTextPrimary-131"
                      >
                        TK2
                      </p>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-spacing-xs-1-24 MuiGrid-item-3 MuiGrid-align-items-xs-center-10 MuiGrid-justify-xs-flex-end-20 MuiGrid-grid-xs-4-39 MuiGrid-grid-sm-6-55"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47 MuiGrid-grid-sm-6-55"
            >
              <div
                autocorrect="off"
                class="MuiInputBase-root-157  makeStyles-input-156 MuiInputBase-fullWidth-167"
                inputmode="decimal"
                spellcheck="false"
              >
                <input
                  aria-label="amount of token to swap"
                  autocomplete="off"
                  class="MuiInputBase-input-168"
                  pattern="^[0-9]*[.,]?[0-9]*$"
                  placeholder="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-175 MuiGrid-container-2 MuiGrid-align-items-xs-center-10 MuiGrid-justify-xs-center-19"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <button
            aria-label="invert tokens swap direction"
            class="MuiButtonBase-root-185 MuiIconButton-root-176"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label-184"
            >
              <svg>
                swap-direction.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root-259"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <div
        class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-104 MuiGrid-container-2 MuiGrid-justify-xs-space-between-21"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <h6
            class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-105 MuiTypography-subtitle1-117 MuiTypography-colorTextPrimary-131"
          >
            To
          </h6>
        </div>
        <div
          class="MuiGrid-root-1 MuiGrid-item-3"
        >
          <h6
            class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-136 MuiTypography-subtitle1-117 MuiTypography-colorTextSecondary-132"
          />
        </div>
      </div>
      <div
        class="MuiBox-root-138 MuiBox-root-188 Styled(MuiBox)-root-137"
      >
        <div
          class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-align-items-xs-center-10"
        >
          <div
            aria-label="token to be swapped"
            class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
          >
            <div
              class="MuiGrid-root-1 MuiGrid-item-3"
            >
              <div
                class="MuiBox-root-138 MuiBox-root-189 Styled(MuiBox)-root-140"
              >
                <div
                  class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-spacing-xs-1-24 MuiGrid-align-items-xs-center-10"
                >
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3"
                  >
                    <div
                      class="MuiAvatar-root-144 MuiAvatar-circle-146 WithStyles(ForwardRef(Avatar))-root-142 WithStyles(ForwardRef(Avatar))-root-190 MuiAvatar-colorDefault-145"
                      dimension="48"
                    >
                      <img
                        alt="token icon"
                        src="token.svg"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-true-35"
                  >
                    <div
                      class="MuiBox-root-138 MuiBox-root-192 Styled(MuiBox)-root-152 Styled(MuiBox)-root-191"
                    >
                      <p
                        class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-155 MuiTypography-body1-108 MuiTypography-colorTextPrimary-131"
                      >
                        TK1
                      </p>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root-1 MuiGrid-item-3"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-1 MuiGrid-container-2 MuiGrid-item-3 MuiGrid-align-items-xs-center-10 MuiGrid-grid-xs-4-39 MuiGrid-grid-sm-6-55"
          >
            <div
              autocorrect="off"
              class="MuiInputBase-root-157  makeStyles-input-156 MuiInputBase-fullWidth-167"
              inputmode="decimal"
              spellcheck="false"
            >
              <input
                aria-label="amount of token to be swapped"
                autocomplete="off"
                class="MuiInputBase-input-168"
                pattern="^[0-9]*[.,]?[0-9]*$"
                placeholder="0"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root-1 WithStyles(ForwardRef(Grid))-root-193 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <h6
        class="MuiTypography-root-106 WithStyles(ForwardRef(Typography))-root-136 Styled(WithStyles(ForwardRef(Typography)))-root-194 MuiTypography-subtitle1-117 MuiTypography-colorTextSecondary-132"
      >
        <span>
          1 TK2 = 
        </span>
        <span>
          <span
            class="MuiSkeleton-root-196 MuiSkeleton-text-197 WithStyles(ForwardRef(Skeleton))-root-195 MuiSkeleton-pulse-200"
            style="width: 20px;"
          />
        </span>
        <span>
           TK1
        </span>
      </h6>
    </div>
    <div
      class="MuiGrid-root-1 MuiGrid-item-3 MuiGrid-grid-xs-12-47"
    >
      <button
        class="MuiButtonBase-root-185 MuiButton-root-208 MuiButton-contained-216 WithStyles(ForwardRef(Button))-root-207 MuiButton-containedSecondary-218 MuiButton-disableElevation-219 MuiButton-disabled-221 MuiButton-fullWidth-231 MuiButtonBase-disabled-186"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label-209"
        >
          Enter an amount
        </span>
      </button>
    </div>
  </div>
</div>
`;
